src/os/data/filedescriptor.js

```diff
@@ -173,25 +177,41 @@ os.data.FileDescriptor.prototype.getLayerOptions = function() {
 
 
 /**
- * Get the column mappings to apply to imported data.
- *
- * @return {Array.<os.im.mapping.IMapping>}
+ * @inheritDoc
  */
 os.data.FileDescriptor.prototype.getMappings = function() {
   return this.parserConfig['mappings'];
 };
 
 
 /**
- * Set the column mappings to apply to imported data.
- *
- * @param {Array.<os.im.mapping.IMapping>} value
+ * @inheritDoc
  */
 os.data.FileDescriptor.prototype.setMappings = function(value) {
   this.parserConfig['mappings'] = value;
```

Comment:
`setMappings` was previously only called when the descriptor was created. Now that we're also calling it after creation, we need to make sure the descriptor gets saved to settings. Currently if you change ellipse mappings on an active file layer and refresh, your changes are lost.

Reply:
So that's what I have `updateMappings()` for, actually saving the descriptor for refresh (not sure if that's the same as saving to settings). I wanted to keep this function as simple as possible since it's also used in `geometrystep.js` for import and then call `updateMappings()` anytime someone adds a new mapping after import.

Reply:
Ok, that makes sense. We should make sure the descriptor gets saved in `updateMappings` then.